internal/commands/build.go

```diff
@@ -41,10 +45,31 @@ func Build(logger logging.Logger, cfg config.Config, packClient PackClient) *cob
 				suggestSettingBuilder(logger, packClient)
 				return MakeSoftError()
 			}
-			env, err := parseEnv(flags.EnvFiles, flags.Env)
+
+			descriptor, err := parseProjectToml(flags.AppPath, flags.DescriptorPath)
+			if err != nil {
+				return err
+			}
+
+			env, err := parseEnv(descriptor, flags.EnvFiles, flags.Env)
 			if err != nil {
 				return err
 			}
+
+			buildpacks := flags.Buildpacks
+			if len(buildpacks) == 0 {
+				buildpacks = []string{}
+				for _, bp := range descriptor.Build.Buildpacks {
+					if len(bp.Uri) == 0 {
+						// there are several places through out the pack code where the "id@version" format is used.
```

Comment:
We have this as part of a [`BuildpackInfo#FullName()`](https://github.com/buildpacks/pack/blob/master/internal/dist/buildpack.go#L46)

Reply:
That's is where it constructs the string, but there isn't a place were it gets parsed. It didn't look like creating a `BuildpackInfo` in each place was straight forward so i decided not to deal with it in this PR (hence leaving the comment).

Reply:
Theoretically, `project.Buildpack` could embed `dist.BuildpackInfo` and thereby inherit that functionality but I can see how that's not optimal since it would require this package to have a dependency on `dist`. There are additional benefits that come from it as well like ensuring there is only a single location for the definition of what "Buildpack Info" is. Additionally, we use this pattern in multiple places. For example:

https://github.com/buildpacks/pack/blob/589d9975ff9321aa4b9dae66f412708752b259be/internal/builder/metadata.go#L22-L25

Reply:
I don't have any strong opinions on the solution, but i do think it should be done in a separate line of work (hence the `TODO`).

Reply:
Oh, looks like I didn't actually put `TODO`, but you get the idea. It's not the goal of this PR to fix the `BuildpackInfo`.